.github/workflows/paralleltest.yml

```
@@ -6,98 +6,86 @@ on:
     steps:
-      - name: Configuration
-        id: configuration
-        shell: python
-        run: |
-          import os
-          import json
-
-          github_runner = None
-          num_lists = ${{ inputs.testLists }}
-          test_lists = []
-
-          # Choose an agent based on the system being tested default to input parameter if nothing matches
-          if "windows-x64" in '${{ inputs.pool }}' or "windows-arm64" in '${{ inputs.pool }}' or "linux-x64" in '${{ inputs.pool }}' or "linux-arm64" in '${{ inputs.pool }}':
-            github_runner = f"config=['self-hosted', '1ES.Pool=${{ inputs.pool }}']"
-          elif "macos-x64" in '${{ inputs.pool }}':
-            github_runner = "config='macos-11'"
-          elif "macos-arm64" in '${{ inputs.pool }}':
-            github_runner = "config=['self-hosted', 'macOS', 'ARM64']"
-          else:
-            github_runner = "config='${{ inputs.pool }}'"
-
-          with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
-            print(github_runner, file=fh)
-
-          # Generate the test lists
-          for list in range(num_lists):
-            test_lists.append(f"testList_{list}")
+      - name: Checkout
+        uses: actions/checkout@v2
 
-          with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
-            print(f"testConfig={json.dumps(test_lists)}", file=fh)
+      - name: Setup
+        id: setup
+        run: |
+          python ./scripts/vendors/microsoft/configure.py --pool ${{ inputs.pool }} --testLists ${{ inputs.testLists }}
 
   setup-parallel:
+    name: Generate ${{ matrix.suite }} parallelList
     runs-on: ${{ fromJSON(needs.configuration.outputs.agent-config) }}
-    name: Generate AQA parallelList - ${{ inputs.system }}_${{ matrix.suite }}
     needs: configuration
     strategy:
       fail-fast: true
       matrix:
         suite: [functional, openjdk, system, perf]
     steps:
-      - name: Setup windows environment
-        working-directory: ${{ github.workspace }}
-        if: ${{ inputs.system == 'windows' }}
-        shell: powershell
-        run: |
-          curl.exe -o windows-binaries.zip -L "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}?${{ secrets.SAS_TOKEN }}"
-          Expand-Archive .\windows-binaries.zip .\tmp
-          $jdk_directory = Get-ChildItem .\tmp\
-          $jdk_directory = $jdk_directory.Basename
-          $jdk_directory = "${{ github.workspace }}\tmp\$jdk_directory"
-          echo "TEST_JDK_HOME=$jdk_directory" >> $env:GITHUB_ENV
-
-      - name: Setup non-windows environment
-        working-directory: ${{ github.workspace }}
-        if: ${{ inputs.system != 'windows' }}
-        shell: bash
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Download artifacts
+        uses: ./.github/vendor-templates/microsoft/az_download_artifacts
+        with:
+          blobs: '${{ inputs.customizedSdkUrl }},${{ inputs.testImage }}'
```

I think we can send this as an array like:
```suggestion
          blobs: 
            - ${{ inputs.customizedSdkUrl }}
            - ${{ inputs.testImage }}
```

What do you think?

---

Ah, I didnt know that syntax was available for GHA. I can make the changes. I prefer this way since its more explicit.

---

Ah, I do not think that we are able to set these this way. I committed the changes but locally I see errors on those lines. Might need to send them in comma separated for now.